### PR TITLE
Add unsubscribe

### PR DESCRIPTION
**Pull request by harmonr**

### CODE REVIEW

**Comment by djarufes** [comment]
image

File: src/app/groups/groups.component.html [hunk 1]
```diff
@@ -30,9 +30,26 @@
       </button>
     </div>
     <div class="col align-self-center">
-      <button mat-raised-button color="primary" (click)="onJoin(group)">
-        Join
-      </button>
+      <div *ngIf="subscribedGroups.includes(group.id); else join">
+        <button
+          mat-raised-button
+          class="btn-custom"
```
Comment (djarufes) [inline src/app/groups/groups.component.html:36]: btn-custom could be changed to w-25 to get almost the same thing. w-25 makes the button a bit wider, but we get rid of the css

File: src/app/chat-list-item/chat-list-item.component.ts [hunk 2]
```diff
@@ -33,7 +33,7 @@ export class ChatListItemComponent implements OnInit {
       this.imageUrl = data?.groupImageUrl;
       this.chatTitle = data?.groupName;
       /* Subscribe to the recipient only once by checking the isRecipientSubscribed */
-      if (!this.isRecipientSubscribed && data.type == 'dm') {
+      if (!this.isRecipientSubscribed && data && data.type == 'dm') {
```
Comment (mburakyucel) [inline src/app/chat-list-item/chat-list-item.component.ts:36]: Why do we check if data is valid? Doesn't it return a chatMetadata on success response?

Comment (harmonr) [reply src/app/chat-list-item/chat-list-item.component.ts:36]: I am getting an error when first starting up the app, if this is fine then it is not needed

Comment (harmonr) [reply src/app/chat-list-item/chat-list-item.component.ts]: Changed it back, and tested again, no errors. Do not know why it gave me an error before.

**Review by mburakyucel — APPROVED** [review]

**Pull request merged**